_rules/focusable-element-has-visible-focus-f4e323.md

```diff
@@ -0,0 +1,352 @@
+---
+id: f4e323
+name: Focusable element has visible focus
+rule_type: atomic
+description: |
+  This rule checks that each focusable element has visible focus indication
+accessibility_requirements: # Remove whatever is not applicable
+  wcag20:2.4.7: # Focus Visible (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: satisfied
+input_aspects:
+  - CSS styling
+  - DOM tree
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+---
+
+## Applicability
+
+The rule applies to any [focusable][] element.
+
+## Expectation
+
+For each target element, there exists a set of [focus indicators][focus indicator] which are not all also [focus indicators][potential focus indicator] for another [focusable][] element (the same for all the indicators).
```

Comment:
I think you're going to have a very hard time to come up with a complete list of all the different ways something can be focusable. Perhaps focus indicator can be defined as all pixels that change when the element is focused from when the element is not focused.

You could add a second expectation that says each focus indicator must be unique, to replicate this "for another focusable element" thing, but I'd much prefer to have that as an assumption. I don't think I've ever seen that happen, so that seems like a hypothetical we shouldn't spend our time on.

Reply:
I disagree with the need for unicity. Check Passed Example 6 where each element has 2 focus indicator (one on each side) and these focus indicators are shared between two elements. The only unicity condition that applies to this example is that the set of stuff that change is unique.